apps/ejabberd/src/mod_mam_utils.erl

```diff
@@ -227,38 +233,51 @@ external_binary_to_mess_id(BExtMessID) when is_binary(BExtMessID) ->
 %% -----------------------------------------------------------------------
 %% XML
 
+%% @doc Adds all arcid elements (for backward compatibility)
```

Comment:
This is a first iteration of this task. Some deployments may rely on `archived` element, so in a second iteration we'll make it configurable which element should be added.